content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md

```diff
@@ -0,0 +1,428 @@
+3. [Usernetes](https://github.com/rootless-containers/usernetes)
+
+### Immutable container filesystems
+
+The Kubernetes Hardening Guidance report highlights an often overlooked feature `readOnlyRootFileSystem`, with a
+working example in Appendix B. This example limits execution and tampering of
+containers at runtime. Any read/write activity can then be limited to few
+directories by using `tmpfs` volume mounts.
+
+However, for some applications that modify container filesystem at runtime like 
+exploding a WAR or JAR file at container start up could face issues when 
+enabling this feature. To avoid this issue, it is important to consider making 
+minimal changes to filesystem at runtime when possible.
+
+### Building secure container images
+
+Kubernetes Hardening Guidance also recommends running a scanner at deploy time as an admission controller,
+to prevent vulnerable or misconfigured pods from running in the cluster.
+Theoretically, this sounds like a good approach but there are several caveats to
+consider before this can be implemented in practice:
+
+- Depending on network bandwidth, available resources and scanner of choice,
+  scanning for vulnerabilities for an image can take an indeterminate amount of
+  time. This could lead to slower or unpredictable pod start up times, which
+  could result in spikes of unavailability when apps are serving peak load.
+- If the policy that allows or denies pod startup is made using incorrect or
+  incomplete data it could result in several false positive or false negative
+  outcomes like this:
+  - inside a container image, the `openssl` package is detected as vulnerable. However,
+    the application is written in Golang and uses the Go `crypto` package for TLS. Therefore, this vulnerability
+    is not in the code execution path and as such has minimal impact if it
+    remains unfixed
+  - A vulnerability is detected in the `openssl` package for a Debian base image.
+    However, the upstream Debian community considers this as a Minor impact
+    vulnerability and as a result does not release a patch fix for this
+    vulnerability. Owner of this image is now stuck with a vulnerability that
+    can not be fixed and a cluster that does not allow the image to run because
+    of predefined policy that does not take into account whether the fix for a
+    vulnerability is available or not
+  - A Golang app is built on top of a [distroless](https://github.com/GoogleContainerTools/distroless)
+    image, but it is compiled with a Golang version that uses a vulnerable [standard library](https://pkg.go.dev/std).
+    The scanner has
+    no visibility into golang version but only on OS level packages. So it 
+    allows the pod to run in the cluster in spite of the image containing an 
+    app binary built on vulnerable golang.
+
+To be clear, relying on vulnerability scanners is absolutely a good idea but
```

Comment:
Suggest dropping the "To be clear" verbiage, it doesn't convey anything in particular
```suggestion
Relying on vulnerability scanners is absolutely a good idea but
```